cmake/StormSetup.cmake

```diff
@@ -1,10 +1,10 @@
 cmake_minimum_required(VERSION 3.12.0)
 
 macro(_get_relative_path_folder out path)
-    list(APPEND base_folders ${PUBLIC_INCLUDE_DIRS} ${SRC_DIRS})
+    set(base_folders "${PUBLIC_INCLUDE_DIRS};${SRC_DIRS}")
     list(JOIN base_folders "|" folders_match)
     get_filename_component(dir ${path} DIRECTORY)
-    string(REGEX REPLACE "(.*${CMAKE_CURRENT_SOURCE_DIR}/(${folders_match})\/?)" "" ${out} "${dir}")
+    file(RELATIVE_PATH ${out} "${CMAKE_CURRENT_SOURCE_DIR}" "${dir}")
```

Comment:
Breaks VS filters. Is there any reason why we should avoid `REGEX REPLACE`?

Reply:
Files paths with `.`, `+`, `(` or `)`. 

The `+` was causing issues for me. So we could still use regexps but then we would need to escape the current source directory first.

I made a mistake with the excluding the `base_folders` though, will try to fix that.

Reply:
Ah, I see! I didn't think about it before
But quick googling says we can prepare the path part of that regex with `string(REGEX REPLACE "([][+.*()^])" "\\\\\\1" regex "${string}")`

Reply:
I combined the relative path with the regex search. Should work now, as long as we don't put strange folder names in the project (which we shouldn't).